components/ui-api-layer/docs/proposals/exceed-resource-quota.md

```diff
@@ -0,0 +1,126 @@
+# Checking when resource quota has been exceeded proposal
+
+## Overview
+
+We have a few ideas how to check if resource quota exceeded its limit.
+
+## Checking for conditions
+
+We can use deployment's or replica set's conditions to judge if given resource quota has been exceeded.
+
+These are the conditions from:
+Replica set: 
+```yaml
+
+{
+     "lastTransitionTime": "2018-08-16T09:41:52Z",
+     "message": "pods \"redis-client-6d75bbc666-lrvg6\" is forbidden: exceeded quota: kyma-default, requested: limits.memory=512Mi, used: limits.memory=3Gi, limited: limits.memory=3Gi",
+     "reason": "FailedCreate",
+     "status": "True",
+     "type": "ReplicaFailure"
+}
+
+```
+Deployment:
+```yaml
+{
+     "lastTransitionTime": "2018-08-16T09:41:52Z",
+     "lastUpdateTime": "2018-08-16T09:41:52Z",
+     "message": "pods \"redis-client-6d75bbc666-lrvg6\" is forbidden: exceeded quota: kyma-default, requested: limits.memory=512Mi, used: limits.memory=3Gi, limited: limits.memory=3Gi",
+     "reason": "FailedCreate",
+     "status": "True",
+     "type": "ReplicaFailure"
+}
+
+```
+When ui-api would get a call for resource usage in the given namespace, it would list all resources and check their conditions, if they contains such a (failed) condition.
```

Comment:
``When the ui-api-layer gets a call for the resource usage in a given Namespace, it lists all resources and checks if they have any `failed` condition.``

Reply:
Done